### ISSUE TITLE
Include NoSidewalk label in analysis

### ISSUE THREAD

**Issue by misaugstad** [manaswis/cmsc634_sidewalk-mTurk#11]
We didn't include this label in the analysis for our class project. It will take some thought, as the NoSidewalk label doesn't have to be placed at specific positions.

How do we decide what a true/false positive/negative is? Do we just aggregate over a street segment, so a NoSidewalk label is only associated with street segments as far as analysis is concerned?

**Comment by maddalihanumateja** [comment]
Aggregating over a street edge seems like a good way to go about this

**Comment by misaugstad** [comment]
Yeah I do think I like this, as a street segment sort of approximates a city on average, and what you really want to know is if there is no sidewalk on some block.

For actually using the tool, you want more information than this, since one side of the street missing a sidewalk doesn't matter so much if the other side of the street has it. The method above may be sufficient for just calculating accuracy, however.

**Comment by misaugstad** [comment]
A note about what Jon had said at a meeting about this: He mentioned that one way he envisions using the NoSidewalk label is to label at every single step, and then a line can be formed from the first label placed by someone up to the last label placed by them, and this gives a fairly accurate indication of where sidewalk is actually missing.

This leads to the question... what is ground truth and what is accuracy?

Some ways that lines could be compared: length and distance between ground truth endpoint & turkers enpoint. And assuming a turker line segment overlaps with a ground truth line segment (so it is a sub-segment..?), then the percentage of the length can be used to calculate accuracies...

For example: the dataset has just one NoSidewalk segment (in the GT), and it is 100m long. Then the turkers end up with a NoSidewalk segment that is only 85m long, but is a sub-segment of the GT NoSidewalk segment. Then maybe their "recall" is 85% and their "precision" is 100%. Similarly, if GT was 85m long and turkers were 100m long, then "recall" would be 100% and "precision" would be 85%.
